src/hexkit/providers/s3/testutils/_fixtures.py

```diff
@@ -0,0 +1,302 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Utilities for testing code that uses the S3ObjectStorage provider.
+
+Please note, only use for testing purposes.
+"""
+
+import os
+from collections.abc import AsyncGenerator, Generator
+from contextlib import contextmanager
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import Any, NamedTuple, Optional
+
+try:
+    from typing import Self
+except ImportError:  # Python < 3.11
+    from typing_extensions import Self
+
+import pytest
+import pytest_asyncio
+from pydantic import SecretStr
+from testcontainers.localstack import LocalStackContainer
+
+from hexkit.custom_types import PytestScope
+from hexkit.providers.s3.provider import S3Config, S3ObjectStorage
+
+from ._utils import FileObject, calc_md5, populate_storage, upload_file, upload_part
+
+__all__ = [
+    "LOCALSTACK_IMAGE",
+    "MEBIBYTE",
+    "TEST_FILE_DIR",
+    "TEST_FILE_PATHS",
+    "upload_file",
+    "populate_storage",
+    "calc_md5",
+    "FileObject",
+    "S3ContainerFixture",
+    "get_s3_container_fixture",
+    "s3_container_fixture",
+    "S3Fixture",
+    "get_persistent_s3_fixture",
+    "persistent_s3_fixture",
+    "get_clean_s3_fixture",
+    "clean_s3_fixture",
+    "s3_fixture",
+    "tmp_file",
+    "temp_file_object",
+]
+
+
+LOCALSTACK_IMAGE = "localstack/localstack:3.4.0"
```

Comment:
```suggestion
LOCALSTACK_IMAGE = "localstack/localstack:3.5.0"
```

We can use this PR to update the image.

Reply:
Good point. Fixed this and the below comments